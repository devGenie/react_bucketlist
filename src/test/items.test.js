import React from 'react';
import { mount, shallow } from 'enzyme';
import { stub, spy } from 'sinon';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import 'should';
import 'should-enzyme';
import localStorageMock from '../__mocks__/localStorage';
import bucketlistmock from '../__mocks__/bucketlists-mock';
import BucketListItems from '../components/bucketlist-items';
import BucketListItem from '../components/bucketlist-item'
import sinonStubPromise from 'sinon-stub-promise';

sinonStubPromise(sinon)
window.sessionStorage = localStorageMock;

describe("<BucketListItems/>",()=>{
	let itemdata={id:1,name:'Bucket1',complete_status:true,date_completed:'mm/yy'};
	let bucketlistitems;
	let doneFunction;
	let editFuncion;
	let deleteFunction;
	let editFunc;
	let fetchMock;
	let item;

	beforeEach(()=>{
		editFunc = stub();
		fetchMock = stub(window,'fetch').returnsPromise().resolves({});
		doneFunction=spy(BucketListItem.prototype,'handleDone');
		editFuncion=spy(BucketListItem.prototype,'handleEdit');
		deleteFunction = spy(BucketListItem.prototype,'handleDelete');
		bucketlistitems = mount(<BucketListItems editFunc={editFunc}
	 										   data={[itemdata]}/>);
	 	item = mount(<BucketListItem data={itemdata}/>);
	})

	afterEach(()=>{
		fetchMock.restore();
		doneFunction.restore();
		editFuncion.restore();
		deleteFunction.restore();
	})
	
	 it("can render successfully",()=>{
	 	expect(toJson(bucketlistitems)).toMatchSnapshot()
	 })

	 it("item can render successfully",()=>{
	 	expect(toJson(bucketlistitems)).toMatchSnapshot()
	 })

	 it("handle item can be done",()=>{
	 	let done = bucketlistitems.find('.items').find('.completeButton').simulate('click');
	 	expect(doneFunction.called).toBe(true)
	 })

	 it("handle item can be deleted",()=>{
	 	bucketlistitems.find('.items').find('.deleteButton').simulate('click');
	 	expect(deleteFunction.called).toBe(true)
	 })

	 it("can call editFunc prop",()=>{
	 	bucketlistitems.find('.items').find('.editButton').simulate('click');
	 	console.log(bucketlistitems.find('.items').find('.editButton').length)
	 	expect(editFunc.called).toBe(true)
	 })

	 it("Renders items",()=>{
	 	expect(bucketlistitems.find('.items')).toHaveLength(1);
	 })
});
