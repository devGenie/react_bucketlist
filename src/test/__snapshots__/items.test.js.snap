// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BucketListItems/> can render successfully 1`] = `
<BucketListItems
  data={
    Array [
      Object {
        "complete_status": true,
        "date_completed": "mm/yy",
        "id": 1,
        "name": "Bucket1",
      },
    ]
  }
  editFunc={[Function]}
>
  <ul
    className="bkitems"
  >
    <BucketListItem
      data={
        Object {
          "complete_status": true,
          "date_completed": "mm/yy",
          "id": 1,
          "name": "Bucket1",
        }
      }
      itemIndex={0}
      key="0"
      onEdit={[Function]}
    >
      <li
        className="items"
      >
        <span
          className="item_name"
        >
          Bucket1
        </span>
        <div
          className="right"
        >
          <SeeMore>
            <div
              className="over_flow_button"
            >
              <a
                className="dropdown_toggle"
              >
                <i
                  className="material-icons"
                >
                  more_vert
                </i>
              </a>
              <div
                className="overflow_content"
              >
                <a
                  className="dropdown_icon completeButton"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons"
                  >
                    check_box
                  </i>
                </a>
                <a
                  className="dropdown_icon editButton"
                  data-target="item_edit_model"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons"
                  >
                    edit
                  </i>
                </a>
                <a
                  className="dropdown_icon deleteButton"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons"
                  >
                    delete
                  </i>
                </a>
              </div>
            </div>
          </SeeMore>
        </div>
        <span
          className="complete_status"
        >
          <div
            className="success"
          >
            <span
              className="left"
            >
              Completed
            </span>
            <i
              className="material-icons right"
            >
              check_box
            </i>
          </div>
        </span>
      </li>
    </BucketListItem>
  </ul>
</BucketListItems>
`;

exports[`<BucketListItems/> item can render successfully 1`] = `
<BucketListItems
  data={
    Array [
      Object {
        "complete_status": true,
        "date_completed": "mm/yy",
        "id": 1,
        "name": "Bucket1",
      },
    ]
  }
  editFunc={[Function]}
>
  <ul
    className="bkitems"
  >
    <BucketListItem
      data={
        Object {
          "complete_status": true,
          "date_completed": "mm/yy",
          "id": 1,
          "name": "Bucket1",
        }
      }
      itemIndex={0}
      key="0"
      onEdit={[Function]}
    >
      <li
        className="items"
      >
        <span
          className="item_name"
        >
          Bucket1
        </span>
        <div
          className="right"
        >
          <SeeMore>
            <div
              className="over_flow_button"
            >
              <a
                className="dropdown_toggle"
              >
                <i
                  className="material-icons"
                >
                  more_vert
                </i>
              </a>
              <div
                className="overflow_content"
              >
                <a
                  className="dropdown_icon completeButton"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons"
                  >
                    check_box
                  </i>
                </a>
                <a
                  className="dropdown_icon editButton"
                  data-target="item_edit_model"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons"
                  >
                    edit
                  </i>
                </a>
                <a
                  className="dropdown_icon deleteButton"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons"
                  >
                    delete
                  </i>
                </a>
              </div>
            </div>
          </SeeMore>
        </div>
        <span
          className="complete_status"
        >
          <div
            className="success"
          >
            <span
              className="left"
            >
              Completed
            </span>
            <i
              className="material-icons right"
            >
              check_box
            </i>
          </div>
        </span>
      </li>
    </BucketListItem>
  </ul>
</BucketListItems>
`;
